### ISSUE TITLE
Refactor ngResizeObserverProviders to be compatible with Angular libraries (ng-packagr) / AOT

### ISSUE THREAD

**Issue by gthau** [ChristianKohler/ng-resize-observer#2]
An Angular library that uses ng-resize-observer cannot be built.
This is because the ngResizeObserverProviders uses a function expression. Function expression cannot be used in decorators when building with AOT.

```
ERROR: ng-resize-observer/ng-resize-observer.ts(73,15): Error during template compile of 'ɵb'
  Function expressions are not supported in decorators
    Consider changing the function expression into an exported function.
```

https://github.com/ChristianKohler/ng-resize-observer/blob/9cdc5d0e72430e87451838dabddada75b9b5066e/projects/ng-resize-observer/src/lib/ng-resize-observer.service.ts#L73

As suggested by the error message, the function expression has to be declared as an exported function such as:
```ts
// doesn't work with AOT
export const NgResizeObserverProvider = {
  provide: NgResizeObserver,
  useFactory: (resizeObserverService: ResizeObserverService) =>
    resizeObserverService.onResize,
  deps: [ResizeObserverService]
};

// should work with AOT
export function ngResizeObserverFactory(resizeObserverService: ResizeObserverService) {
  return resizeObserverService.onResize;
}

export const NgResizeObserverProvider = {
  provide: NgResizeObserver,
  useFactory: ngResizeObserverFactory,
  deps: [ResizeObserverService]
};
```

See this gist for additional details:
https://gist.github.com/chuckjaz/65dcc2fd5f4f5463e492ed0cb93bca60#supported-expression-subset

> Notable ommisions from this list include function expressions and lambdas. Including a function expression or a lambda in an annotation will report an error reqesting the lambda or function expression be turned into an exported function

**Comment by ChristianKohler** [comment]
Thank you @gthau 

Totally missed that. Fixed it with version 1.0.1

**Comment by gthau** [comment]
Thank you for the useful library and the timely fix :)
